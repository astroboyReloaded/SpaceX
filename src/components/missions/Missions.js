import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../../redux/missions/missionsSlice';
import MissionBadge from './MissionBadge';
import JoinBtn from './JoinBtn';
import style from '../../styles/missions.module.css';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions, loading, error } = useSelector((state) => state.missions);

  useEffect(() => {
    if (missions.length) return;
    dispatch(fetchMissions());
  }, [missions, dispatch]);

  return (
    <>
      {loading && <span>loading...</span>}
      {missions.length > 0 && (
        <table className={style.table}>
          <tr className={style.tableRow}>
            <th className={style.tableData}>Mission</th>
            <th className={style.tableData}>Description</th>
            <th className={style.tableData}>Status</th>
          </tr>
          {missions.map((mission) => (
            <tr className={style.tableRow} key={mission.id}>
              <td className={`${style.tableData} ${style.name}`}>{mission.name}</td>
              <td className={style.tableData}>{mission.description}</td>
              <td className={style.tableData}>
                <MissionBadge joined={mission.joined} />
              </td>
              <td className={style.tableData}>
                <JoinBtn
                  id={mission.id}
                  joined={mission.joined}
                />
              </td>
            </tr>
          ))}
        </table>
      )}
      {error && <span>error</span>}
    </>
  );
};

export default Missions;
