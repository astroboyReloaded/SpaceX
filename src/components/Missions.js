import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions, handleMission } from '../redux/missions/missionsSlice';
import Badge from './Badge';
import ReserveBtn from './ReserveBtn';

const Missions = () => {
  const dispatch = useDispatch();
  const { missions } = useSelector((state) => state.missions);
  console.log(missions);

  useEffect(() => {
    if (missions.length) return;
    dispatch(fetchMissions());
  }, [missions, dispatch]);

  return (
    <>
      <table>
        <tr>
          <th>Mission</th>
          <th>description</th>
          <th>Status</th>
        </tr>
        {missions.map((mission) => (
          <tr key={mission.id}>
            <td>{mission.name}</td>
            <td>{mission.description}</td>
            <td>
              <Badge
                reserved={mission.reserved}
                label={mission.reserved ? 'Active Member' : 'NOT A MEMBER'}
                type="mission"
              />
            </td>
            <td>
              <ReserveBtn
                id={mission.id}
                reserved={mission.reserved}
                handleReservation={handleMission}
                label={mission.reserved ? 'Leave Mission' : 'Join Mission'}
              />
            </td>
          </tr>
        ))}
      </table>
    </>
  );
};

export default Missions;
