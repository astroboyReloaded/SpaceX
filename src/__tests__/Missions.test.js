import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../redux/Store';
import Missions from '../components/missions/Missions';

it('Renders Missions', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Routes>
            <Route path="/missions" element={<Missions />} />
          </Routes>
        </Router>
      </Provider>,
    ).toJSON();
  expect(tree).toMatchSnapshot();
});
