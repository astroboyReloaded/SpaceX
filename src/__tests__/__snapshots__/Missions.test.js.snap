// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Missions 1`] = `null`;
